tests/erlang_tests/test_lists_keyfind.erl

```diff
@@ -0,0 +1,71 @@
+-module(test_lists_keyfind).
+
+-export([start/0]).
+
+start() ->
+    ok = test_keyfind_with_existing_key(),
+    ok = test_keyfind_with_non_existing_key(),
+    ok = test_keyfind_with_empty_list(),
+    ok = test_keyfind_with_existing_key_on_different_position(),
+    ok = test_keyfind_with_existing_key_and_different_length_of_tuples(),
+    ok = test_keyfind_with_invalid_input(),
+    0.
+
+test_keyfind_with_existing_key() ->
+    Key = bob,
+    Tuple1 = {1, alice},
+    Tuple2 = {2, bob},
+    Tuple3 = {3, carol},
+    List = [Tuple1, Tuple2, Tuple3],
+    {2, bob} = lists:keyfind(Key, 2, List),
```

Comment:
I suggest wrapping at least `List` with some kind of `id` external function for future-proofing against compiler optimizations. I don't think this is going to happen soon, however `lists` module is quite crucial (and a good target for future optimizations).  This comment applies to all other test files that have been introduced here.